testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java

```diff
@@ -955,6 +956,69 @@ void highConcurrencyOnSingleConnection() {
     pool.shutdown();
   }
 
+  @Test
+  void spanEndsAfterBodyReceived() throws Exception {
+    assumeTrue(options.isSpanEndsAfterBody());
+
+    String method = "GET";
+    URI uri = resolveAddress("/long-request");
+
+    int responseCode =
+        doRequest(
+            method,
+            uri,
+            // the time that server waits before completing the response
+            Collections.singletonMap("delay", String.valueOf(TimeUnit.SECONDS.toMillis(1))));
+
+    assertThat(responseCode).isEqualTo(200);
+
+    testing.waitAndAssertTraces(
+        trace -> {
+          trace.hasSpansSatisfyingExactly(
+              span -> assertClientSpan(span, uri, method, 200, null).hasNoParent(),
+              span -> assertServerSpan(span).hasParent(trace.getSpan(0)));
+          SpanData span = trace.getSpan(0);
+          // make sure the span is at least as long as the delay we set when sending the request
+          assertThat(
+                  span.getEndEpochNanos() - span.getStartEpochNanos()
+                      >= TimeUnit.SECONDS.toNanos(1))
```

Comment:
It looks like we are setting a delay of 1 millisecond, but asserting that the span is longer than 1 nanosecond. Should those match?

Reply:
Perhaps you misread this `TimeUnit.SECONDS.toNanos(1)` returns 1 second in nanos = 1000000000